tests/test_schema.py

```diff
@@ -197,3 +197,37 @@ def test_normalize_std(x):
 def test_normalize_error():
     with pytest.raises(SchemaError):
         assert normalize_field_type('thingy')
+
+
+@requires_only_gdal1
+def test_check_schema_driver_support(tmpdir):
+
+    # Shapefile
+    for field_type in ['datetime', 'time']:
+        try:
+            name = str(tmpdir.join('test_scheme.shp'))
+            items = [('field1',  field_type)]
+            with fiona.open(name, 'w',
+                            driver="ESRI Shapefile",
+                            schema={
+                                'geometry': 'LineString',
+                                'properties': items}) as c:
+                assert False
+            
+        except DriverSupportError:
+            assert True
```

Comment:
@rbuffat would you be willing to change the above in two ways? 1) the several field types could be done using pytest.mark.parametrize https://docs.pytest.org/en/latest/parametrize.html, which creates separate tests for each field type, and 2) use `with pytest.raises(DriverSupportError)` to assert that an exception is raised.

Reply:
I refactored the test using pytest.raises and parametrize